### PR TITLE
Add EventService CRUD methods

### PR DESCRIPTION
**Pull request by jeffplays2005**
Added the following:
* Create event
* Get event by ID
* Update event
* Delete event by ID

### CODE REVIEW

**Review by bcho892 — COMMENTED** [review]
LGTM, some simple tests would be good

File: server/src/data-layer/services/EventService.ts [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+   */
+  public async createEvent(event: Event) {
+    return await FirestoreCollections.events.add({
+      ...event,
+      start_date: Timestamp.fromMillis(
+        event.start_date.seconds * 1000 + event.start_date.nanoseconds / 1000000
+      ), // Need to do this because firestore does not like "raw" {seconds, nanoseconds}
+      end_date: Timestamp.fromMillis(
+        event.end_date.seconds * 1000 + event.end_date.nanoseconds / 1000000
+      ) // Need to do this because firestore does not like "raw" {seconds, nanoseconds}
```
Comment (bcho892) [inline server/src/data-layer/services/EventService.ts:20]: this should be done in the tests, not here

File: server/src/data-layer/services/EventService.ts [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+   * Gets an existing reservation document by ID.
+   *
+   * @param eventId the ID of the event document
+   * @param reservationId the ID of the reservation document
+   * @returns the reservation document
+   */
+  public async getReservation(eventId: string, reservationId: string) {
+    const result = await FirestoreCollections.events
+      .doc(eventId)
+      .collection("reservations") // subject to place as a constant somewhere
```
Comment (bcho892) [inline server/src/data-layer/services/EventService.ts:83]: yea this should be fine - can probably create something like `FirestoreSubcollections` where we can obtain the reservations for just that event, given an `eventId`


i.e give event id `uid` -> `FirestoreSubcollections.reservations(uid)`

File: server/src/data-layer/services/EventService.test.ts
Comment (bcho892) [inline server/src/data-layer/services/EventService.test.ts:1]: do we want to check for all reservations being deleted with parent?

**Review by bcho892 — APPROVED** [review]

File: server/src/data-layer/services/EventService.test.ts [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+    )
+    expect(fetchedReservation1).toBe(undefined)
+    const fetchedReservation2 = await eventService.getReservation(
+      newEvent.id,
+      newReservation2.id
+    )
+    expect(fetchedReservation2).toBe(undefined)
+  })
+
+  it("Should not delete other reservations when deleting an event document", async () => {
```
Comment (bcho892) [inline server/src/data-layer/services/EventService.test.ts:122]: :+1:

File: server/src/data-layer/adapters/Firestore.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    return admin.firestore()
+  },
+  {
+    doc: <T>(path: string) => {
+      return admin.firestore().doc(path).withConverter<T>(converter<T>())
+    },
+    collection: <T>(path: string) => {
+      return admin.firestore().collection(path).withConverter<T>(converter<T>())
+    },
+    subcollection: <T>(path: string, docId: string, subpath: string) => {
```
Comment (bcho892) [inline server/src/data-layer/adapters/Firestore.ts:19]: nit: jsdoc would be helpful for this

**Review by bcho892 — APPROVED** [review]

**Comment by jeffplays2005** [comment]
Added jsdoc!

**Pull request merged**